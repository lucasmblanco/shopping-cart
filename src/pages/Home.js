import React from 'react'
import ImageOne from '../assets/images/home/home-1.jpg'; 
import ImageTwo from '../assets/images/home/home-2.jpg';
import ImageThree from '../assets/images/home/home-3.jpg'; 
import { Link } from 'react-router-dom';

export default function Home() {
  return (
    <div className='h-[calc(100vh_-_15rem)] text-3xl font-inter flex justify-content'>
      <div className='absolute top-image-one-top left-image-one-left w-image-width  rounded-xl overflow-hidden z-10'>
        <img  src={ImageOne} alt='promo'></img>
      </div>
      <div className='absolute top-80 left-image-three-left w-image-width  rounded-xl overflow-hidden z-0'>
        <img src={ImageThree} alt="promo-three"/>
      </div>
      <div className='absolute top-72 left-image-two-left w-image-width rounded-xl overflow-hidden '>
        <img src={ImageTwo} alt="promo two"/>
      </div>
      <div className='h-44 w-div-home-width bg-red-accent rounded-xl z-20 text-left'>
        <p className='m-3 w-50'>WE MAKE THE FUTURE AVAILABLE</p>
        <div className='flex justify-between p-3 content-center'>
          <div className='flex items-center text-white ' >TODAY</div>
          <Link to='/shop' className='bg-black rounded-xl text-white p-3'>SHOW NOW</Link>
          </div>

      </div>
    </div>
  )
}
