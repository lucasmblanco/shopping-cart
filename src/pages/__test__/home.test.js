import React from "react";
import { render, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import '@testing-library/jest-dom';
import Home from '../Home';

test('renders Home component', () => {
  const { asFragment } = render(<Home />, { wrapper: MemoryRouter });

  expect(asFragment()).toMatchSnapshot();
});

test('should render three images elements', () => {
  render(<Home />, { wrapper: MemoryRouter });

  const images = screen.getAllByRole('img');

  images.forEach((img) => expect(img).toBeInTheDocument());
});

test('should render div component with a p element and span element', () => {
  render(<Home />, { wrapper: MemoryRouter });

  screen.getByText('WE MAKE THE AVAILABLE');
  screen.getByText('FUTURE');
}); 

