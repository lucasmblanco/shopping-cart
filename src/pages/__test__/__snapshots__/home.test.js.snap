// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Home component 1`] = `
<DocumentFragment>
  <div
    class="relative h-[calc(100vh_-_14rem)] text-3xl font-inter grid grid-cols-home grid-rows-home-rows z-0"
    data-testid="home-component"
  >
    <img
      alt="promo"
      class="top-image-one-top
 left-image-one-left w-image-width
 rounded-xl overflow-hidden z-10
 col-start-10 col-span-6 row-start-12
 row-span-18 shadow-xl animate-fade-in-bck"
      src="home-1.jpg"
    />
    <img
      alt="promo"
      class="top-image-one-top
 left-image-one-left w-image-width
 rounded-xl overflow-hidden z-0
 col-start-14 col-span-6 row-start-3
 row-span-18 shadow-xl animate-fade-in-bck-2"
      src="home-2.jpg"
    />
    <img
      alt="promo"
      class="top-image-one-top
 left-image-one-left w-image-width
 rounded-xl overflow-hidden z-0
 col-start-6 col-span-6 row-start-3
 row-span-18 shadow-xl animate-fade-in-bck-3"
      src="home-3.jpg"
    />
    <div
      class="h-44 w-div-home-width bg-red-accent rounded-xl z-20 col-start-15
 row-start-14 shadow-xl border-solid border-light-red-accent border-2 animate-fade-in-bck-4"
    >
      <p
        class="m-3 w-50"
      >
        WE MAKE THE 
        <span
          class="text-white-accent"
        >
          FUTURE
        </span>
         AVAILABLE
      </p>
      <div
        class="flex justify-evenly content-center"
      >
        <div
          class="flex items-center text-white-accent"
        >
          TODAY
        </div>
        <a
          class="bg-black rounded-xl text-white-accent p-3
 shadow-xl transition-all duration-500 hover:bg-light-red-accent"
          href="/shop"
        >
          SHOP NOW
        </a>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
